cdap-app-templates/cdap-etl/cdap-etl-lib/src/main/java/co/cask/cdap/templates/etl/batch/sources/PartitionedFileSetSource.java

```
@@ -0,0 +1,127 @@
+                                       PartitionedFileSetProperties.builder()
+        .setPartitioning(ETLUtils.createPartitioning(stageConfig.getProperties().get(PARTITIONING)))
+        .setInputFormat(TextInputFormat.class)
+        .setOutputFormat(TextOutputFormat.class)
+        .setOutputProperty(TextOutputFormat.SEPERATOR, separator)
+        .build());
+    }
+  }
+
+  @Override
+  public void prepareJob(BatchSourceContext context) {
+    Map<String, String> inputArgs = Maps.newHashMap();
+    PartitionedFileSetArguments.setInputPartitionFilter(inputArgs,
+                                                        createPartitionFilter(
+                                                          context.getRuntimeArguments().get(FILTER)));
+    PartitionedFileSet input = context.getDataset(context.getRuntimeArguments().get(FILESET_NAME), inputArgs);
+    context.setInput(context.getRuntimeArguments().get(FILESET_NAME), input);
+  }
+
+  /**
+   * creates a {@link PartitionFilter} from a given string in JSON format
+   *
+   * @param filterString partition filter JSON string
+   * @return {@link PartitionFilter}
+   */
+  private PartitionFilter createPartitionFilter(String filterString) {
+    PartitionFilter.Builder builder = PartitionFilter.builder();
+    Type stringStringMap = new TypeToken<Map<String, String>>() {
+    }.getType();
+    Map<String, String> filters = new Gson().fromJson(filterString, stringStringMap);
+    for (Map.Entry<String, String> entrySet : filters.entrySet()) {
+      String key = entrySet.getKey();
+      if (key.endsWith(VALUE)) {
+        builder.addValueCondition(key.split("\\.")[0], entrySet.getValue());
+      } else if (key.endsWith(LOWER)) {
+        String fieldName = key.split("\\.")[0];
+        // if the upper range is found for  this lowe range add it to PartitionFilter else just ignore
+        String upperRange = Joiner.on(".").join(fieldName, UPPER);
+        if (filters.containsKey(upperRange)) {
```

seems like we should error out if there is no upper range instead of silently ignoring it